site/en/docs/extensions/mv3/architecture-overview/index.md

```diff
@@ -2,370 +2,107 @@
+[The popup][doc-popup]
+: Many extensions use a popup to provide functionality, such as displaying a list of tabs, or additional information regarding the current tab. Users can easily find it by clicking on the extension toolbar icon. When the user navigates away it will automatically close.
 
-``` js
-let image = chrome.runtime.getURL("images/my_image.png")
-```
+[The options page][doc-options]
+: The options page provides a way for users to customize an extension; such as choosing which sites the extension will run on. Users can access the options page in several ways as described in [Finding the options page][doc-options-view].
 
-To access an extension file from a **website**, you will have to construct the URL as follows:
+Other extension HTML pages include [Chrome override pages][doc-override], [sandbox pages][doc-sandbox] or any custom page included for a specific purpose like onboarding the user.
 
-```text
-chrome-extension://EXTENSION_ID/RELATIVE_PATH
-```
+### Other assets {: #assets }
 
-You can find the <code><var>EXTENSION_ID</var></code> in the Extension management page
-**chrome://extensions**. The <code><var>RELATIVE_PATH</var></code> is the file path relative to the
-extension's top folder.
+An extension can include many types of resources, but only the [extension icons][manifest-icons] are required. All assets, including images and fonts, must be part of the extension package.
 
-{% Aside 'key-term' %}
+## How they work together {: #interact }
 
-The **extension ID** is a 32-character alpha string that identifies an extension in the browser and
-on the Chrome Web Store.
+In this section, we will describe how these extension components communicate, store data, and share access to resources.
 
-{% endAside %}
+### Sending messages {: #pageComm }
 
-During development, a new ID is generated when an [_unpacked extension_][docs-unpacked] is loaded,
-unless the `"key"` property is [set in the manifest][docs-key].
+Many times content scripts, or other extension pages, need to send or receive information from the extension service worker. In these cases, either side can listen for messages sent from the other end, and respond on the same channel. Extensions can send a one-time request or establish a long-lived connection to support multiple messages.
 
-{% Aside 'caution' %}
+See [Message passing][doc-messages] for more details.
 
-All assets that content scripts and websites want to access must be declared under
-[`web_accessible_resources`][section-web-res] key in the manifest.
+### Storing data {: #data }
 
-{% endAside %}
+Chrome provides extensions with an optimized [Storage API][api-storage] available to all extension components. It includes four separate storage areas for specific use cases and an event listener that tracks whenever data is updated. For example, when you save changes in the popup, the extension service worker can respond with specified logic.
```

Comment:
"Optimized" has implications that may not be accurate. It would be safer to say that the Storage API is purpose-built, specialized, or designed for extensions. 

```suggestion
Chrome provides extensions with a specialized [Storage API][api-storage] available to all extension components. It includes four separate storage areas for specific use cases and an event listener that tracks whenever data is updated. For example, when you save changes in the popup, the extension service worker can respond with specified logic.
```